src/cider/nrepl/middleware/info.clj

```diff
@@ -87,7 +87,7 @@
                 (when (= "file" (.getProtocol rsrc))
                   (io/as-file rsrc))
                 (io/as-file filename))]
-    (when (and file (.exists file))
+    (when (and file (.exists file) (.isFile file))
```

Comment:
What problem did you run in to with the existing code?

If `file` isn't a `File` then the order of the statements in the `and` should be re-ordered or you risk running `.exists` on a String or whatever else `file` might be.